File: src/main/java/com/architecture_map/belarus/service/SourceService.java
package com.architecture_map.belarus.service;

import com.architecture_map.belarus.dto.SourceDto;
import com.architecture_map.belarus.entity.Source;

import java.util.List;
import java.util.Optional;

public interface SourceService {

    Source create(SourceDto sourceDto);

    List<Source> findAll();

    Optional<Source> updateById(Integer id, SourceDto sourceUpdates);

    Optional<Source> patchUpdateById(Integer id, SourceDto sourceDto);

    boolean deleteByid(Integer id);
}

File: src/main/kotlin/by/architecturemap/belarus/service/SourceService.kt
package by.architecturemap.belarus.service

import by.architecturemap.belarus.dto.SourceDTO
import by.architecturemap.belarus.entity.Source

interface SourceService {

    fun create(source: Source): Source
    fun find(id: Int): Source
    fun findAll(): List<Source>
    fun update(id: Int, updatedSource: Source): Source
    fun patchUpdate(id: Int, updatedSource: SourceDTO): Source
    fun delete(id: Int)
}
